src/auth/integration-tests/tests/driver.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[cfg(all(test, feature = "run-integration-tests"))]
+mod driver {
+    use auth::credentials::create_access_token_credential;
+    use gax::error::Error;
+    use gax::options::ClientConfig as Config;
+    use scoped_env::ScopedEnv;
+    use secretmanager::client::SecretManagerService;
+
+    type Result<T> = std::result::Result<T, gax::error::Error>;
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+    #[serial_test::serial]
+    async fn service_account() -> Result<()> {
```

Comment:
nit: in the other integration tests we keep the bulk of the code outside the `#[cfg(all(test, feature = ...]` to always compile and lint the code, even if we do not run it.  Consider doing the same here.

Reply:
Ack, thanks. I did not understand why the other tests were written that way.

I started making this change. The change is significant enough to require another review.

I will merge this commit first in the name of incremental progress.